templates/kubernetes-cluster.template

```diff
@@ -376,29 +391,22 @@ Resources:
             kubeadm reset
 
             # Initialize master node
-            kubeadm init --cloud-provider=aws --token=${ClusterToken}
+            kubeadm init \
+              --cloud-provider=aws \
+              --token=${ClusterToken} \
+              --api-external-dns-names=${LoadBalancerHostname} \
+              --use-kubernetes-version=v1.5.2
 
             # Add-on for network Calico
             # (http://docs.projectcalico.org/v2.0/getting-started/kubernetes/installation/hosted/kubeadm/)
             # so pods can communicate
             kubectl apply -f https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}/scripts/${NetworkingProvider}.yaml
 
-            # Expose the api server over ELB
-            kubectl -n kube-system expose pod \
-              kube-apiserver-$(hostname -f) \
-              --type=LoadBalancer \
-              --port 443 \
-              --target-port=6443 \
-              --name apiserver-public
-
-            # Wait until it's available
-            set +e
-            while :; do
-              LB_HOSTNAME=$(kubectl -n kube-system get svc apiserver-public -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
-              if [[ $? -eq 0 ]]; then break; fi
-              sleep 1
-            done
-            set -e
+            # Add this machine (master) to the load balancer for external access
+            sudo aws elb register-instances-with-load-balancer \
+              --load-balancer-name ${LoadBalancerName} \
+              --instances $(ec2metadata --instance-id) \
```

Comment:
If ec2metadata fails it'll be hidden, I think and may have some strange failure modes.  Probably better to either grab it into its own variable above.